### PR TITLE
Lighting & Shaders Improvements

### PR DESCRIPTION
**Pull request by raphydaphy**
Made several improvements to lighting and shaders
- The highest tile for every x position in the world is now tracked (nothing uses this yet but it has many uses)
- Lighting now penetrates much less into solid blocks, creating a nicer cave ambience (see RockBottomGame/RockBottom#19)
- The raw world coordinates are now sent to the shaders and translated on the GPU
- Scaling is now entirely handled by shaders

Also check the [API](https://github.com/RockBottomGame/API/pull/34) and [Assets](https://github.com/RockBottomGame/Assets/pull/21) PR's.

### CODE REVIEW

File: src/main/java/de/ellpeck/rockbottom/render/WorldRenderer.java [hunk 1]
```diff
@@ -101,19 +101,32 @@ public void calcCameraValues(IRenderer g) {
...
+    private IShaderProgram setCameraValues(IAssetManager manager, ResourceName shader, float transX, float transY, float scale) {
+        IShaderProgram program = manager.getShaderProgram(shader);
+        program.bind();
+        program.setUniform("camera", transX, transY, scale);
+        return program;
+    }
+
     public void render(IGameInstance game, IAssetManager manager, ParticleManager particles, IRenderer g, AbstractWorld world, EntityPlayer player, InteractionManager input) {
-        float scale = g.getWorldScale();
+        calcCameraValues(manager, g);
```
Comment (Quarris) [inline src/main/java/de/ellpeck/rockbottom/render/WorldRenderer.java]: this dot

File: src/main/java/de/ellpeck/rockbottom/world/tile/TileLog.java [hunk 2]
```diff
@@ -48,6 +48,12 @@ public void onDestroyed(IWorld world, int x, int y, Entity destroyer, TileLayer
         }
     }
 
+    public float getTranslucentModifier(IWorld world, int x, int y, TileLayer layer, boolean skylight) {
+        if (world.getState(layer, x, y).get(StaticTileProps.LOG_VARIANT).isNatural())
+            return skylight ? 1F : layer == TileLayer.BACKGROUND ? 0.9f : 0.8f;
```
Comment (Quarris) [inline src/main/java/de/ellpeck/rockbottom/world/tile/TileLog.java]: no. separate the conditions

Comment (raphydaphy) [reply src/main/java/de/ellpeck/rockbottom/world/tile/TileLog.java]: u literally watched me type this but ok

Comment (Quarris) [reply src/main/java/de/ellpeck/rockbottom/world/tile/TileLog.java]: yes, and both myself and ellpeck clearly laughed at you and said to change it :P

**Comment by Ellpeck** [comment]
merge this, it is good

**Pull request closed without merging**